gamemode/rounds.lua

```diff
@@ -0,0 +1,98 @@
+--- A round lasts from everyone spawning, until two players each of a different team have fought to the death or finished for over a minute (or 1 player if playing solo).
+
+--- The round we're playing at
+GM.Round = 1
+
+--- The amount of seconds between two players finishing and the round forcibly restarting
+GM.RoundEndGraceConVar = CreateConVar("slide_round_end_grace", 60, nil, "The time between two players of different teams finishing and the round ending.", 1)
+
+local STALLING_TIMER_NAME = "PreventRoundStalling"
+
+--- Make a player join a round
+--- @param ply GPlayer
+function GM:JoinRound(ply)
+  ply.RoundNumber = self.Round
+end
+
+--- Resets a round, cleaning up the map and allowing everybody to respawn
+function GM:ResetRound()
+  local newRoundNumber = self.Round + 1
+  PrintMessage(HUD_PRINTTALK, string.format("Starting new round (#%s)!", newRoundNumber))
+
+  timer.Remove(STALLING_TIMER_NAME)
+
+  -- Kill all other players silently
+  for _, ply in pairs(player.GetAll()) do
+    if(IsValid(ply) and ply:Alive())then
+      ply:KillSilent()
+    end
+  end
+
+  -- Do this in a timer, so we don't remove an invalid entity (like the skull that spawns on death, or ragdoll??)
+  -- Prevents a crash by doing game.CleanUpMap in a timer, unsure why
+  timer.Simple(0, function()
+    -- Quick workaround for issue @ https://github.com/Facepunch/garrysmod-issues/issues/3637
+    game.CleanUpMap(false, {"env_fire", "entityflame", "_firesmoke"})
```

Comment:
There are very few maps that actually need this called on them regularly, just ones like collapse. A couple have destructible bits that you only really want to clean up every 5-10 minutes and a couple have one time doors that can be mapfixed to re-open after 30 seconds.

That said, I'm not sure even collapse needs this called every single round.